src/Contributor/ContributorRepository.php

```diff
@@ -55,24 +55,27 @@ public function isContributor(UUID $id, EmailAddress $emailAddress): bool
         return count($results) > 0;
     }
 
-    public function addContributor(UUID $id, EmailAddress $emailAddress): void
+    public function overwriteContributors(UUID $id, EmailAddresses $emailAddresses)
     {
-        $this->connection
-            ->insert(
-                self::TABLE,
-                [
-                    'uuid' => $id->toString(),
-                    'email' => $emailAddress->toString(),
-                ]
-            );
-    }
-
-    public function deleteContributors(UUID $id): void
-    {
-        $this->connection
-            ->delete(
-                self::TABLE,
-                ['uuid' => $id->toString()]
-            );
+        $this->connection->transactional(
```

Comment:
Oh good idea, the transaction is a nice added benefit of doing it in the repository. I hadn't even considered that 👍